### PR TITLE
Add citation links to the GitHub repo

### PR DESCRIPTION
**Pull request by tzanio**
GitHub [recently added](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-citation-files#citing-something-other-than-software) the option to show citation info for repositories based on a `CITATION.cff` file.

This PR such a file for MFEM pointing to the CAMWA paper.

You can see the resulting link on the right of https://github.com/mfem/mfem/tree/citation-dev.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2481](https://github.com/mfem/mfem/pull/2481) | @tzanio | @tzanio | @pazner + @v-dobrev | 08/19/21 | 08/19/21 | 08/19/21 | |

### CODE REVIEW

**Comment by tzanio** [comment]
Folks, can you please take a look and update your ORCID numbers?

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    orcid: "https://orcid.org/0000-0000-0000-0000"
+  - family-names: "Cerveny"
+    given-names: "Jakub"
+    orcid: "https://orcid.org/0000-0000-0000-0000"
+  - family-names: "Dobrev"
+    given-names: "Veselin"
+    orcid: "https://orcid.org/0000-0000-0000-0000"
+  - family-names: "Dudouit"
+    given-names: "Yohann"
+    orcid: "https://orcid.org/0000-0000-0000-0000"
```
Comment (YohannDudouit) [inline CITATION.cff:34]:
```suggestion
    orcid: "https://orcid.org/0000-0001-5831-561X"
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  - family-names: "Zampini"
+    given-names: "Stefano"
+    orcid: "https://orcid.org/0000-0000-0000-0000"
+  doi: "10.1016/j.camwa.2020.06.009"
+  journal: "Computers and Mathematics with Applications"
+  month: 1
+  start: 42 # First page number
+  end: 74 # Last page number
+  title: "MFEM: A Modular Finite Element Library"
```
Comment (pazner) [inline CITATION.cff:67]:
```suggestion
  title: "MFEM: A Modular Finite Element Methods Library"
```

**Comment by tzanio** [comment]
I also need two approvals :)

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    orcid: "https://orcid.org/0000-0000-0000-0000"
+  - family-names: "Dahm"
+    given-names: "Johann"
+    orcid: "https://orcid.org/0000-0000-0000-0000"
+  - family-names: "Medina"
+    given-names: "David"
+    orcid: "https://orcid.org/0000-0000-0000-0000"
+  - family-names: "Zampini"
+    given-names: "Stefano"
+    orcid: "https://orcid.org/0000-0000-0000-0000"
```
Comment (stefanozampini) [inline CITATION.cff:61]:
```suggestion
    orcid: "https://orcid.org/0000-0002-0435-0433"
```

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    given-names: "Johann"
+    orcid: "https://orcid.org/0000-0000-0000-0000"
+  - family-names: "Medina"
+    given-names: "David"
+    orcid: "https://orcid.org/0000-0000-0000-0000"
+  - family-names: "Zampini"
+    given-names: "Stefano"
+    orcid: "https://orcid.org/0000-0002-0435-0433"
+  doi: "10.1016/j.camwa.2020.06.009"
+  journal: "Computers and Mathematics with Applications"
```
Comment (pazner) [inline CITATION.cff:63]:
```suggestion
  journal: "Computers & Mathematics with Applications"
```
We should check that this doesn't mess up the BibTeX rendering.

Comment (tzanio) [reply CITATION.cff:63]: Try it here: https://github.com/tzanio/mfem

Comment (tzanio) [reply CITATION.cff:63]: With `Computers & Mathematics with Applications` the BibTeX is:

```bib
@article{Anderson_MFEM_A_Modular_2021,
author = {Anderson, Robert and Andrej, Julian and Barker, Andrew and Bramwell, Jamie and Camier, Jean-Sylvain and Cerveny, Jakub and Dobrev, Veselin and Dudouit, Yohann and Fisher, Aaron and Kolev, Tzanio and Pazner, Will and Stowell, Mark and Tomov, Vladimir and Akkerman, Ido and Dahm, Johann and Medina, David and Zampini, Stefano},
doi = {10.1016/j.camwa.2020.06.009},
journal = {Computers & Mathematics with Applications},
month = {1},
pages = {42--74},
title = {{MFEM: A Modular Finite Element Library}},
volume = {81},
year = {2021}
}
```

With `Computers \& Mathematics with Applications` I get an error.

Maybe there is another way of doing it?

Comment (pazner) [reply CITATION.cff:63]: Yeah, it doesn't escape the `&` unfortunately ☹️ 

`Computers \& Mathematics with Applications` should work fine (it does for me)

Comment (tzanio) [reply CITATION.cff:63]: Like this: https://github.com/tzanio/mfem/blob/master/CITATION.cff ?

Comment (tzanio) [reply CITATION.cff:63]: I got it... double escape

Comment (pazner) [reply CITATION.cff:63]: Oh, sorry, I misunderstood. `\&` works for the BibTeX, but I couldn't get it to work with the `cff` file.

Lets just leave it as it is then 🙂

Comment (tzanio) [reply CITATION.cff]: it is fixed :)

Comment (pazner) [reply CITATION.cff]: Smart!!

**Review by pazner — APPROVED** [review]

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by mlstowell — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**